vue.d.ts has bad reference to module 'v3-component-public-instance'

### Version
2.7.6

### Reproduction link
[dev.azure.com](https://dev.azure.com/definitelytyped/DefinitelyTyped/_build/results?buildId=134865&view=logs&j=6399d89d-3d30-5fff-3e92-07bbaf7ebe2f&t=482eae7c-85d2-5e8a-ea70-a5ab41e9bc13&l=12436)







### Steps to reproduce
Install vue@2.7.6 and try to compile with typescript:

```
$ mkdir fresh; cd fresh
$ npm i vue@2.7.6
$ echo "import * as v from 'vue'" >index.ts
$ tsc index.ts --target esnext --moduleResolution node
```

### What is expected?
No errors

### What is actually happening?
Error from tsc:
node_modules/vue/types/vue.d.ts(17,41): error TS2307: Cannot find module 'v3-component-public-instance' or its corresponding type declarations.


---
Found in various vue 2 dependents on Definitely Typed's overnight test run.

'v3-component-public-instance' is a local module, so the reference should be './v3-component-public-instance'

When editing vue.d.ts inside the vue repo, tsconfig is set up such that 'v3-component-public-instance' *also* resolves, but this doesn't work when vue is imported normally.

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/types/tsconfig.json
+++ b/types/tsconfig.json
@@ -9,7 +9,6 @@
     "jsx": "preserve",
     "strict": true,
     "noEmit": true,
-    "baseUrl": ".",
     "paths": {
       "vue": ["../index.d.ts"]
     }

--- a/types/vue.d.ts
+++ b/types/vue.d.ts
@@ -14,7 +14,7 @@ import { VNode, VNodeData, VNodeChildren, NormalizedScopedSlot } from './vnode'
 import { PluginFunction, PluginObject } from './plugin'
 import { DefineComponent } from './v3-define-component'
 import { nextTick } from './v3-generated'
-import { ComponentPublicInstance } from 'v3-component-public-instance'
+import { ComponentPublicInstance } from './v3-component-public-instance'
 
 export interface CreateElement {
   (

